File: exercises/c-sharp/apprentice-bootcamp-fundamentals-2/apprentice-bootcamp-fundamentals-2/Receipt.cs
namespace apprentice_bootcamp_fundamentals_2 {
  using System;

  public class Receipt {
    private string _text = String.Empty;
    private int _total;
    private int _numberOfA;
    private int _numberOfB;

    public string Text {
      get => $"{this._text}Total: {this._total}";
    }

    public void ScannedA() {
      this._text = $"{this._text}A: 50";
      this._numberOfA++;
      this._total += 50;
      if (this._numberOfA % 3 == 0) {
        this._text = $"{this._text} - 20 (3 for 130)";
        this._total -= 20;
      }
      this._text = $"{this._text}\n";
    }

    public void ScannedB() {
      this._text = $"{this._text}B: 30";
      this._total += 30;
      this._numberOfB++;
      if (this._numberOfB % 2 == 0) {
        this._text = $"{this._text} - 15 (2 for 45)";
        this._total -= 15;
      }
      this._text = $"{this._text}\n";
    }

    public void ScannedC() {
      this._text = $"{this._text}C: 20\n";
      this._total += 20;
    }

    public void ScannedD() {
      this._text = $"{this._text}D: 15\n";
      this._total += 15;
    }
  }
}

File: exercises/javascript/main/checkout/receipt.js
class Receipt {
  constructor() {
    this.text = '';
    this.numberOfA = 0;
    this.numberOfB = 0;
    this.total = 0;
  }

  getText() {
    return this.text + "Total: " + this.total;
  }

  scannedA() {
    this.text += 'A: 50';
    if (++this.numberOfA % 3 == 0) {
      this.text += ' - 20 (3 for 130)';
      this.total += 30;
    } else {
      this.total += 50;
    }
    this.text += '\n';
  };

  scannedB() {
    this.text += 'B: 30';
    if (++this.numberOfB % 2 == 0) {
      this.text += ' - 15 (2 for 45)';
      this.total += 15;
    } else {
      this.total += 30;
    }
    this.text += '\n';
  };

  scannedC() {
    this.text += 'C: 20\n';
    this.total += 20;
  };

  scannedD() {
    this.text += 'D: 15\n';
    this.total += 15;
  };
};

module.exports = {
  Receipt
};